src/main/java/org/wildfly/naming/client/AbstractContext.java

```diff
@@ -327,7 +327,7 @@ public NameParser getNameParser(final String name) throws NamingException {
         return getNativeNameParser();
     }
 
-    Name decomposeName(CompositeName compositeName) throws NamingException {
+    protected Name decomposeName(CompositeName compositeName) throws NamingException {
```

Comment:
If this method is made available, then line 332 would have to return ``parser.parse("")`` instead of ``new SimpleName();`` I _think_.  This is in the event that a different class form is used by the context by default.